tasks/tomcat_wars.yml

```diff
@@ -0,0 +1,29 @@
+---
+- name: "Ensure webapp dir exists (url)"
+  shell: "mkdir -p {{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps"
+  when:
+    - "tomcat_wars.url is defined"
+  with_items: "{{ tomcat_wars.url }}"
+- name: "loop over tomcat_wars"
+  get_url:
+    dest: "{{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps/"
+    url: "{{ item }}"
+    validate_certs: false
+  when:
+    - "tomcat_wars.url is defined"
+  with_items: "{{ tomcat_wars.url }}"
+- name: "Ensure webapp dir exists (s3)"
+  shell: "mkdir -p {{ item.directory | default(tomcat_directory) }}/{{ tomcat_archive[item.version | default (tomcat_version)]['dir'] }}/webapps"
```

Comment:
Again: the module `file` would be better. Just to explain; shell is `changed` all the time, while `file` is idempotent.

Reply:
mkdir -p is idempotent, but I'll change it. :)